canary/models/Anomaly_Explanation.md

```diff
@@ -0,0 +1,365 @@
+
+# Anomaly Explanation
+
+
+```python
+import json
+from collections import defaultdict
+import os
+from glob import glob
+import pickle
+from copy import deepcopy
+
+import pandas as pd
+import numpy as np
+from matplotlib import pyplot as plt
+import seaborn as sns
+
+from canary.generator.utils import X_metric_to_df, day_bucket_to_points, y_metric_to_df
+from canary.models.utils import (
+    percentile, plot_dists, plot_preds, _calculate_one_day_dist
+)
+from canary.models.metrics import (
+    log_bhattacharyya_dist, log_max_diff_dist, log_sum_of_diff_dist, log_wasserstein_dist, log_energy_dist
+)
+from canary.generator.utils import buckets_to_points
+```
+
+    Using TensorFlow backend.
+
+
+## Load the data
+
+Loading the data previously generated with `generator` module. 
+ - The `SPLIT_DATA_DIR` should contain data generated by `generate_anomalies_in_split_data.py`. 
+ - The `DATA_DIR` should contain data generated with `generate_test_data.py`.
+ - The `PREDICTIONS_DIR` should contain the predictions of the Statistical Model that can be produced with `Statistical.ipynb`.
+
+
+```python
+SPLIT_DATA_DIR = '/home/aga/canary/canary/generator/'
+DATA_DIR = '/home/aga/canary/data/'
+PREDICTIONS_DIR = '/home/aga/canary/data/'
+```
+
+
+```python
+# TODO: Zmienic na with open
```

Comment:
What's this TODO about?

Reply:
Oh, sorry, it's about using `with open()` instead of open